src/main/java/sparksoniq/semantics/types/AtomicType.java

```diff
@@ -0,0 +1,31 @@
+package sparksoniq.semantics.types;
+
+
+import java.io.Serializable;
+
+public class AtomicType implements Serializable {
```

Comment:
The name is misleading, as integer? for example is a sequence type, but not an atomic type. An atomic type is a specific kind of item type that does not have a cardinality.

-> Something along the lines of AtomicSequenceType, or CastType (or any other idea you may have)?

Reply:
Thanks! SingleType sounds good.

In Zorba, it is probably something we forgot, that's a good catch. This is really an almost useless corner case and we didn't sweep in all corners. What I want to avoid above all, here, is to open a can of worms if we start treating null differently than other atomic types. Otherwise it will come back to bite us later.

In JSound too, null is a singleton-space atomic type, which we can union with other types to create "integer-or-null" types or the like.